jetty-core/jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/StdErrAppender.java

```diff
@@ -199,9 +201,16 @@ private String renderedLevel(Level level)
         }
     }
 
-    private void appendCause(StringBuilder builder, Throwable cause, String indent)
+    private void appendCause(StringBuilder builder, Set<Throwable> visited, Throwable cause, String indent)
     {
         builder.append(EOL).append(indent);
+        if (visited.contains(cause))
+        {
+            builder.append("[CIRCULAR REFERENCE: ").append(cause.getClass().getName()).append("]");
```

Comment:
Like the JDK, I would append here the whole exception, so call `appendEscaped(builder, cause.toString)` instead of just the class name.

Reply:
Done.